backend/tests/security/test_materialized_views_security.py

```
@@ -0,0 +1,661 @@
+class TestMaterializedViewsSecurity:
+    """Security tests for materialized views"""
+
+    def test_refresh_requires_authentication(self, client):
+        """Test that refresh endpoint requires authentication"""
+        # Execute - No authentication header
+        response = client.post("/materialized-views/refresh", json={})
+
+        # Assert - Should return 401 or 403
+        assert response.status_code in [401, 403]
+
+    def test_refresh_requires_admin_role(self, client, regular_user):
```

Ruff (0.14.3)

Unused method argument: `regular_user`

(ARG002)